tecs/tecs.cpp

```diff
@@ -359,9 +359,11 @@ void TECS::_update_throttle_setpoint(const float throttle_cruise, const matrix::
 		_last_throttle_setpoint = _throttle_setpoint;
 
 		// Calculate throttle integrator state upper and lower limits with allowance for
-		// 10% throttle saturation to accommodate noise on the demand
-		float integ_state_max = (_throttle_setpoint_max - _throttle_setpoint + 0.1f);
-		float integ_state_min = (_throttle_setpoint_min - _throttle_setpoint - 0.1f);
+		// 10% throttle saturation to accommodate noise on the demand. However, if the
+		// integrator gain is zero, also force integrator state to zero, otherwise it could
+		// stay at a large constant value
+		float integ_state_max = _integrator_gain > 0.001f ? (_throttle_setpoint_max - _throttle_setpoint + 0.1f) : 0.0f;
+		float integ_state_min = _integrator_gain > 0.001f ? (_throttle_setpoint_min - _throttle_setpoint - 0.1f) : 0.0f;
```

Comment:
What about skipping the entire block instead of calculating _throttle_integ_state and then clamping it to zero?

``` C++
if (_integrator_gain > 0.0f) {
    lines ~361-380
} else {
    _throttle_integ_state = 0.0f;
}
```